code/datums/abilities/santa.dm

```diff
@@ -217,8 +217,8 @@
 	cooldown = 30 SECONDS
 
 	cast()
-		var/A
-		A = input("Area to jump to", "TELEPORTATION", A) in get_teleareas()
+		var/list/tele_areas = get_teleareas()
+		var/A = tgui_input_list(src, "Area to jump to", "Teleportation", tele_areas)
```

Comment:
You'll need to nullcheck A.

Reply:
Like this?
Checked the other one and change some ! to use isnull.

Reply:
sure